chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -192,8 +196,24 @@ impl<'a> EvmState for EvmStateImpl<'a> {
 }
 
 impl Ledger {
+    fn validate_transaction_nonce(
+        evm: &Ledger,
+        accounts: &account::Ledger,
+        transaction: &EvmTransaction,
+    ) -> Result<(), Error> {
+        let jor_address = evm.address_mapping.jor_address(&transaction.caller);
+        let expected_nonce = match accounts.get_state(&jor_address) {
+            Ok(account_state) => account_state.evm_state.nonce,
+            Err(LedgerError::NonExistent) => 0,
+            Err(_) => unimplemented!("other errors are not expected"),
+        };
+        (expected_nonce == transaction.nonce)
```

Comment:
nit: simple if would be more readable IMO

Reply:
fixed